base/parse.jl

```diff
@@ -218,6 +218,93 @@ function parse(::Type{T}, s::AbstractString; base::Union{Nothing,Integer} = noth
 end
 
 ## string to float functions ##
+function parsefloat_preamble(s::AbstractString, base::Int, i::Int=1)
+    isempty(s) && throw(ArgumentError("Empty string!"))
+    c = ' '
+    sign = 1
+    while isspace(c)
```

Comment:
Not that this loop will be a problem if `s` consists entirely of spaces.  (You'll get a `BoundsError`.)

Reply:
got it, will add a `!done(i,s)` condition.